codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/protocols/RestXml.kt

```diff
@@ -110,3 +113,12 @@ open class RestXml(private val coreCodegenContext: CoreCodegenContext) : Protoco
 
     override fun serverRouterRuntimeConstructor() = "new_rest_xml_router"
 }
+
+/**
+ * Indicates that a service is expected to send XML where the root element name does not match the modeled member name.
+ */
+class AllowInvalidXmlRoot : AnnotationTrait(ID, Node.objectNode()) {
```

Comment:
When I was cross checking the S3 documentation with the S3 model to try and find other operations that may be broken (which unfortunately turned out to be a pointless endeavor&mdash;they're both wrong), I noticed that some S3 operations use the `@xmlName` trait to correct the wrong root. For example, `GetBucketLocationOutput` has `@xmlName("LocationConstraint")` to fix its root. Should the S3 customization just add that trait to the outputs for these cases rather than the codegen needing to handle a separate synthetic trait?

Reply:
I've broken this work out into a separate issue: #1668